aiohttp/web.py

```diff
@@ -332,140 +332,176 @@ def raise_graceful_exit():
     raise GracefulExit()
 
 
-def _make_server_creators(handler, *, loop, ssl_context,
-                          host, port, path, sock, backlog):
+class Server:
+    def __init__(self, *, host=None, port=None, path=None, sock=None,
+                 shutdown_timeout=60.0, ssl_context=None, backlog=128,
+                 access_log_format=None, access_log=access_logger, loop=None):
+        if loop is None:
+            loop = asyncio.get_event_loop()
+        self.loop = loop
+        self._app = app = Application()
+        make_handler_kwargs = dict()
+        if access_log_format is not None:
+            make_handler_kwargs['access_log_format'] = access_log_format
+        self._handler = app.make_handler(loop=loop, access_log=access_log,
+                                         **make_handler_kwargs)
+
+        self.shutdown_timeout = shutdown_timeout
+        self.scheme = scheme = 'https' if ssl_context else 'http'
+        self.ssl_context = ssl_context
+        self.backlog = backlog
+        self.access_log_format = access_log_format
+        self.access_log = access_log
+        self.base_url = URL('{}://localhost'.format(scheme)).with_port(port)
```

Comment:
`URL.build` was pretty nice.

Reply:
Oops sorry I missed this one 😀 I based my branch on an old version by mistake then I rebased too late when I realized my mistake.